docs/processes/software_package_approval.md

```diff
@@ -0,0 +1,70 @@
+# Software approval policy
+
+To maximise the usability of the secure research environments, we pre-install certain software packages that are deemed broadly useful to a cross section of researchers, thus making them available to all users.
+
+Other software packages which are only of interest to a subset of users can be made available for download from remote package repositories.
+Currently, we support PyPI (Python) and CRAN (R) as remote repositories.
+
+Only a subset of packages that have been vetted to mitigate the risk of introducing malicious or unsound software into the secure environment are made available in this way.
+This page sets out the policy for adding software packages to the approved download list and/or software to the pre-installed software list.
+It also describes the procedure that users of the secure environment should follow to request new additions to the allowlist.
+
+## Background
+
+Given the safeguards afforded by the safe haven environment, and the separation of responsibilities between its constituent resources, the level of risk associated with the introduction of third party software packages is considered low.
+Moreover, access to the environment is carefully controlled and there is a presumption of trust in the individual researchers to whom access is granted.
+
+Nevertheless, the introduction of any software into the safe haven must be considered against the potential risks of:
+
+- approved users having access to data to which they shouldn't (e.g. from data mixing)
+- unapproved users having access to data (e.g. from a data breach)
+- poisoning of data and/or outputs
+- resource misuse (allocation of computational resources for unintended or wasteful purposes).
+
+Such risks may originate unwittingly, from a user who wants to "just get the job done", or from a user, network team member or administrator acting maliciously.
+
+Specific risks which this policy aims to mitigate include:
+
+- package name squatting (allowlisting a similarly-named package instead of the intended one)
+- privilege escalation attacks (enabling a user to gain elevated access permissions).
```

Comment:
It might be worth adding a point here about data ingress. In particular with PyPI as there is no review it would be fairly easy to make a package just containing (a large amount of) data.